pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -37,14 +37,15 @@ func NewOpenShiftClusterDynamicValidator(
 	oc *api.OpenShiftCluster,
 	subscriptionDoc *api.SubscriptionDocument,
 	fpAuthorizer autorest.Authorizer,
-) OpenShiftClusterDynamicValidator {
+) *openShiftClusterDynamicValidator {
```

Comment:
why is this change needed?

Reply:
The validation has to propagate the flag back to the call, so I added a func to the object to return it.  I think the flag is implementation specific and shouldn't be on the interface, that's why I made it this way.

The other approach I thought of was to pass in a pointer and have the func mutate it, but if possible I'd like all validation funcs to be immutable.

Reply:
then `OpenShiftClusterDynamicValidator` is no longer used, right?

Reply:
I think so.  I was also thinking the current way might be just temporary and when the API change is needed I can put the flag on the OpenshiftDoc instead and the changes here can be removed.  Would that be better or should I proceed to removing the `OpenshiftClusterDynamicValidator` for now?

Reply:
Removed in the latest commit.